milli/src/heed_codec/facet/mod.rs

```diff
@@ -25,3 +25,44 @@ pub fn try_split_at(slice: &[u8], mid: usize) -> Option<(&[u8], &[u8])> {
         None
     }
 }
+
+use std::borrow::Cow;
+use std::convert::TryInto;
+
+use crate::{try_split_array_at, DocumentId, FieldId};
+
+pub struct FieldIdCodec;
```

Comment:
I think this implementation is not needed, you can use pre-implemented [Heed types](https://docs.rs/heed-types/latest/heed_types/) to encode primitive types. 😄

Reply:
The Heed types don't implement `BytesEncode`/`BytesDecode`, but I am now using them in the implementation of `FieldCodecId` for clarity :)